linux-kernel: update to 6.13.2

KexyBiscuit opened the pull request:
Topic Description
-----------------

- kernel-tools: update to 6.13.2
    - Synchronize patches from linux-kernel.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update to 6.13.2
    - Add experimental patch to fix Flatpak.
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13.2, current HEAD is 6e0c6028915bd2f9837bd2933f9c84de4769d45b.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update config for 6.13.2-rc2
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: update to 6.13.2-rc2
    - Synchronize patches from linux-kernel.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux+kernel: update to 6.13.2
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update to 6.13.2-rc2
    - Routine patch due diligence for upstream releases.
    - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
    - Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am-shazam.html>.
    - Refresh patch series that have been updated upstream.
    - Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.13, current HEAD is 35ab8ed660a22941ee6db74dc461b5c91b180351.
    - Track Phytium patches at https://github.com/AOSC-Tracking/phytium-linux-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
    - Track Ubuntu patches at https://github.com/AOSC-Tracking/ubuntu-kernel-next-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
    - Track openEuler patches at https://github.com/AOSC-Tracking/openeuler-kernel @ aosc/v6.13, current HEAD is f48dc0c42b8317a1cf69179140c67e6dc8ab1a81.
    - Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is f65a1ce63e2b15af8fb76cdb725bef676e2bf1f9.
    - Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is d40949c20fadb3a0e0a1269c1b0d6bd7dc5c513c.
    - Track linux-surface patches at https://github.com/AOSC-Tracking/linux-surface-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
    - Track Rockchip upstream enablement efforts patches at https://github.com/AOSC-Tracking/rockchip-3588-linux @ aosc/v6.13, current HEAD is 35a506f22b00ba5d97b2744275a3885da91a0f90.
    - Track loongarch-xe patches at https://github.com/AOSC-Tracking/FanFansfan-loongson-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13.2-rc2, current HEAD is 36380050f5025a494a09e8f79a27a232c7214a5b.
    - Add patch to fix DET buffer on AMD GPU.
    - Update patch for Ideapad hotkey LEDs to upstream version.
    - Update patch improving RK3588 HDMI Tx display modes handling.
    - Update patch supporting for mic and audio leds on Ideapad.
    - Add patch to fix boot problems on Zhaoxin KX-7000, caused by the XGETBV instruction.
    - Update patch fixing HCCPARAMS register error for LS7A EHCI.
    - Update patch adding high TMDS clock ratio and scrambling support for dw-hdmi-qp in RK3588.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

Package(s) Affected
-------------------

- kernel-tools: 6.13.2-1
- linux+kernel: 3:6.13.2
- linux-kernel-${__VER}: 6.13.2-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

KexyBiscuit committed:
linux-kernel: update to 6.13.2-rc2

- Routine patch due diligence for upstream releases.
  - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
  - Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am-shazam.html>.
  - Refresh patch series that have been updated upstream.
  - Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.13, current HEAD is 35ab8ed660a22941ee6db74dc461b5c91b180351.
  - Track Phytium patches at https://github.com/AOSC-Tracking/phytium-linux-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
  - Track Ubuntu patches at https://github.com/AOSC-Tracking/ubuntu-kernel-next-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
  - Track openEuler patches at https://github.com/AOSC-Tracking/openeuler-kernel @ aosc/v6.13, current HEAD is f48dc0c42b8317a1cf69179140c67e6dc8ab1a81.
  - Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is f65a1ce63e2b15af8fb76cdb725bef676e2bf1f9.
  - Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is d40949c20fadb3a0e0a1269c1b0d6bd7dc5c513c.
  - Track linux-surface patches at https://github.com/AOSC-Tracking/linux-surface-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
  - Track Rockchip upstream enablement efforts patches at https://github.com/AOSC-Tracking/rockchip-3588-linux @ aosc/v6.13, current HEAD is 35a506f22b00ba5d97b2744275a3885da91a0f90.
  - Track loongarch-xe patches at https://github.com/AOSC-Tracking/FanFansfan-loongson-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
  - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13.2-rc2, current HEAD is 36380050f5025a494a09e8f79a27a232c7214a5b.
- Add patch to fix DET buffer on AMD GPU.
- Update patch for Ideapad hotkey LEDs to upstream version.
- Update patch improving RK3588 HDMI Tx display modes handling.
- Update patch supporting for mic and audio leds on Ideapad.
- Add patch to fix boot problems on Zhaoxin KX-7000, caused by the XGETBV instruction.
- Update patch fixing HCCPARAMS register error for LS7A EHCI.
- Update patch adding high TMDS clock ratio and scrambling support for dw-hdmi-qp in RK3588.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux+kernel: update to 6.13.2

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.13.2-rc2

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: update config for 6.13.2-rc2

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KatyushaScarlet commented:
Works fine on Zhaoxin KX-7000.
Here's the dmesg.
[dmesg-6.13.2-fix-kx-7000.log](https://github.com/user-attachments/files/18707984/dmesg-6.13.2-fix-kx-7000.log)

MingcongBai commented:
Tested good on AMD64 (Panasonic Let's Note CF-SV2), LoongArch (XA61200).

KexyBiscuit committed:
linux-kernel: update to 6.13.2

- Add experimental patch to fix Flatpak.
- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13.2, current HEAD is 6e0c6028915bd2f9837bd2933f9c84de4769d45b.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.13.2

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2 and AMD64 generation 1 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/f69c30f9-f480-48e1-9f56-6b596b616722)
![image](https://github.com/user-attachments/assets/993995ba-82da-4359-beab-c03d418eb5ec)

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.2 build-24409262
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27788.1000) 10.0.27788
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/d70982b8-62b3-4201-b57c-2adbc86a0d50)

KexyBiscuit commented:
Smoke testing passed on LX-2510 with Radeon Pro WX 3100.

![image](https://github.com/user-attachments/assets/bb2b6497-4eda-4e28-9d96-11df868cae98)

KexyBiscuit commented:
Smoke testing passed on XA612A0 with Radeon Pro WX 3100.

![image](https://github.com/user-attachments/assets/8528318e-af1c-49f1-8a24-4e06ba940c61)

KexyBiscuit commented:
Smoke testing passed on XA612A0 with LG110.

![图片](https://github.com/user-attachments/assets/114f86a8-d08b-45c2-9658-edfc92b1de09)

KexyBiscuit commented:
Smoke testing passed on Ampere A110 Raptor.

![image](https://github.com/user-attachments/assets/8bffb349-18c7-418f-a0ca-a772fe22130c)

KexyBiscuit commented:
Smoke testing passed on Armv8-A generation 2 Hyper-V VM.

![image](https://github.com/user-attachments/assets/e5fc66ea-1519-4528-b17a-0f960dbc5523)

KexyBiscuit commented:
Smoke testing passed on Latitude D630.

![image_2025-02-09_18-15-48](https://github.com/user-attachments/assets/e0280adf-3762-49d6-8272-e3b924052a4d)

KexyBiscuit commented:
Smoke testing passed on 战神Z8-KP7S2 with NVIDIA GPU only and NVIDIA DRM KMS disabled.

![图片](https://github.com/user-attachments/assets/2255706d-936e-4e6d-bd06-bdbb47345d10)

KexyBiscuit commented:
Smoke testing passed on 战神Z8-KP7S2 with Optimus on.

![图片](https://github.com/user-attachments/assets/e6ff8c03-51c6-4a80-a421-358070f1fdc3)

KexyBiscuit commented:
Smoke testing passed on RedmiBook 16 (锐龙版).

![图片](https://github.com/user-attachments/assets/cce55956-2a35-409f-9336-44ea8a639f8b)

KexyBiscuit commented:
Smoke testing passed on ROG Ally (2023).

![图片](https://github.com/user-attachments/assets/42132f3c-3a4e-4cb5-a568-75314842d337)

KexyBiscuit commented:
Smoke testing passed on Surface Pro (5th Gen).

![图片](https://github.com/user-attachments/assets/72f0c817-7084-46ba-b713-1801c063b9df)

KexyBiscuit commented:
Building passed on amd64, arm64, loongarch64, mips64el, ppc64el, and riscv64. Smoke testing passed on 9 amd64, 2 arm64, 2 loongarch64, and 1 mips64el test systems.
